File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int i=0;
        int j=0;
        double sum=0;
        double maxi=Integer.MIN_VALUE;
        while(j<nums.length)
        {
            sum+=nums[j];
            if(j-i+1<k)
            {
                j++;
            }
            else if(j-i+1==k)
            {
                maxi=Math.max(sum,maxi);
                sum-=nums[i];
                i++;
                j++;
            }
        }
        return maxi/k;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        i=0
        j=0
        sum=0
        maxi=float('-inf')
        while(j<len(nums)):
            sum+=nums[j]
            if(j-i+1<k):
                j+=1
            elif(j-i+1==k):
                maxi=max(sum,maxi)
                sum-=nums[i]
                j+=1
                i+=1
        return maxi/k